source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -156,20 +156,38 @@ def sort_journeys(resp, journey_order, clockwise):
         resp.journeys.sort(journey_sorter[journey_order](clockwise=clockwise))
 
 
+def compute_car_co2_emission(pb_resp, api_request, instance):
+    car = next((j for j in pb_resp.journeys if helpers.is_car_direct_path(j)), None)
+    if car is None or not car.HasField('co2_emission'):
+        # if there is no car journey found, we request kraken to give us an estimation of
+        # co2 emission
+        co2_estimation = instance.georef.get_car_co2_emission_on_crow_fly(api_request['origin'], api_request['destination'])
+        if co2_estimation:
+            # Assign car_co2_emission into the resp, these value will be exposed in the final result
+            pb_resp.car_co2_emission.value = co2_estimation.value
+            pb_resp.car_co2_emission.unit = co2_estimation.unit
+    else:
+        # Assign car_co2_emission into the resp, these value will be exposed in the final result
+        pb_resp.car_co2_emission.value = car.co2_emission.value
+        pb_resp.car_co2_emission.unit = car.co2_emission.unit
+
+
 def tag_journeys(resp):
     """
-    qualify the journeys
+    tag the journeys
     """
-    car = next((j for j in resp.journeys if helpers.is_car_direct_path(j)), None)
-    if car is None or not car.HasField('co2_emission'):
+    # if there is no available car_co2_emission in resp, no tag will be assigned
+    if not resp.car_co2_emission.value and \
```

Comment:
isn't it very co2 tag specific ?
since we have only those tag it works for the moment, but I think it's counter intuitive, I would have nested the following for loop in the `if resp.car_co2_emission.value and resp.car_co2_emission.unit:` condition for this to be clearer